resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesConf.scala

```diff
@@ -60,7 +62,23 @@ private[spark] case class KubernetesConf[T <: KubernetesRoleSpecificConf](
     roleSecretEnvNamesToKeyRefs: Map[String, String],
     roleEnvs: Map[String, String],
     roleVolumes: Iterable[KubernetesVolumeSpec[_ <: KubernetesVolumeSpecificConf]],
-    sparkFiles: Seq[String]) {
+    sparkFiles: Seq[String],
+    hadoopConfDir: Option[String]) {
+
+  def getHadoopConfigMapName: String = s"$appResourceNamePrefix-hadoop-config"
+
+  def getKRBConfigMapName: String = s"$appResourceNamePrefix-krb5-file"
+
+  def getHadoopStepsOrchestrator : Option[HadoopStepsOrchestrator] = hadoopConfDir.map {
+    hConf => new HadoopStepsOrchestrator(
```

Comment:
nit: `hConf =>` stays together with the `{`; if it doesn't fit, use a full method declaration with enclosing braces.